lib/treelify/acts_as_hierarchy.rb

```diff
@@ -18,8 +18,8 @@ def acts_as_hierarchy
 
     module InstanceMethods
       def memberless_ancestors_for(user)
-        ancestors.joins('LEFT JOIN members on hierarchies.id = members.hierarchy_id')
-                 .where("members.user_id != #{user.id} OR members.id IS NULL")
+        ancestors.joins('LEFT JOIN treelify_members on treelify_hierarchies.id = treelify_members.hierarchy_id')
```

Comment:
I see.

This might be helpful to improve readability:

https://github.com/activerecord-hackery/squeel#joins

But then, you will have extra dependency, then better leave it like it is now.